### PR TITLE
NGINX: Allow customising the server name and additional includes

### PR DESCRIPTION
**Pull request by kierenevans**
This allows for some pre-templating copying/sed in subsequent images without
having to duplicate all of the content, as used in #336

### CODE REVIEW

File: php-nginx/etc/confd/templates/nginx/site.conf.tmpl [hunk 1]
```diff
@@ -21,6 +21,9 @@ server {
     set $access_log_enabled 1;
 
     include /etc/nginx/sites-available/default-*.conf;
+    {{ if ne "" (getenv "WEB_ADDITIONAL_INCLUDES_NAME") }}
+        include /etc/nginx/sites-available/{{ getenv "WEB_ADDITIONAL_INCLUDES_NAME" }}-*.conf;
+    {{ end }}
```
Comment (andytson-inviqa) [inline php-nginx/etc/confd/templates/nginx/site.conf.tmpl]: why can't project's just use the existing `default-*`?

Comment (kierenevans) [inline php-nginx/etc/confd/templates/nginx/site.conf.tmpl]: To be able to use default-* and another thing where the other thing varies when multiple vhosts are involved.

File: php-nginx/etc/confd/templates/nginx/site.conf.tmpl [hunk 2]
```diff
@@ -1,5 +1,5 @@
 server {
-    server_name _;
+    server_name {{ getenv "WEB_SERVER_NAME" }};
```
Comment (andytson-inviqa) [inline php-nginx/etc/confd/templates/nginx/site.conf.tmpl]: what does this functionality give in a single nginx server context? or is it for if adding multiple?

Comment (kierenevans) [inline php-nginx/etc/confd/templates/nginx/site.conf.tmpl]: Only for when there are multiple vhosts.

File: php-nginx/etc/confd/templates/nginx/site.conf.tmpl (hunk 1 above)
Comment (andytson-inviqa) [inline php-nginx/etc/confd/templates/nginx/site.conf.tmpl]: I suppose the question is then what in the default-* isn't applicable to the other vhosts (given things like even ssl configuration is in there. Perhaps some thought as to which files from the base image should be always used (or overridable per server)

File: nginx/README.md [hunk 3]
```diff
@@ -79,6 +80,8 @@ WEB_SSL_SESSION_CACHE | Sets the types and sizes of caches that store session pa
 WEB_SSL_SESSION_TIMEOUT | Specifies a time during which a client may reuse the session parameters. | time | 5m
 WEB_SSL_TRUSTED_CERTIFICATES | The trusted certificates to use for OSCP stapling verification and/or SSL client certificate authentication | absolute filename | 
 NGINX_LOG_FORMAT_NAME | Which log format to use for the access log. Two are currently available: combined (provided by NGINX) or combined_with_x_forwarded_for which logs the whole X-Forwarded-For header. | string | combined
+WEB_INCLUDES | A space separated list of files in /etc/nginx/sites-enabled/ to include. ".conf" will be appended automatically. Globs are accepted. | space separated list of partial file names | default-*
```
Comment (andytson-inviqa) [inline nginx/README.md]: I'm going to merge, but I would like to somehow show that this is rarely otherwise needed.

**Pull request merged**